src/ccl_haloprofile.c

```diff
@@ -0,0 +1,329 @@
+        c_lo = gsl_root_fsolver_x_lower (s);
+        c_hi = gsl_root_fsolver_x_upper (s);
+        rootstatus = gsl_root_test_interval (c_lo, c_hi, 0, 0.0001);
+      }
+    while (rootstatus == GSL_CONTINUE && iter < max_iter);
+
+    gsl_root_fsolver_free (s);
+
+
+    // Check for errors
+    if (rootstatus != GSL_SUCCESS) {
+        ccl_raise_gsl_warning(rootstatus, "ccl_haloprofile.c: solve_cvir():");
+        *status = CCL_ERROR_PROFILE_ROOT;
+        ccl_cosmology_set_status_message(cosmo, "ccl_haloprofile.c: solve_cvir(): Root finding failure\n");
+        return NAN;
+    } else {
+        return cvir;
+    }
+}
+
+// Parameters structure for the Einasto integrand
+typedef struct{
+  ccl_cosmology *cosmo;
+  double alpha, rs;
+} Int_Einasto_Par;
+
+static double integrand_einasto(double r, void *params){
+
+    Int_Einasto_Par *p = (Int_Einasto_Par *)params;
+    double alpha = p->alpha;
+    double rs = p->rs;
+
+    return 4.*M_PI*r*r*exp(-2.*(pow(r/rs,alpha)-1)/alpha);
+}
+
+//integrate einasto profile to get normalization of rhos
+static double integrate_einasto(ccl_cosmology *cosmo, double R, double alpha, double rs, int *status){
+    int qagstatus;
+    double result = 0, eresult;
+    Int_Einasto_Par ipar;
+    gsl_function F;
+    gsl_integration_workspace *w = gsl_integration_workspace_alloc(1000);
+
+    // Structure required for the gsl integration
+    ipar.cosmo = cosmo;
+    ipar.alpha = alpha;
+    ipar.rs = rs;
+    F.function = &integrand_einasto;
+    F.params = &ipar;
+
+    // Actually does the integration
+    qagstatus = gsl_integration_qag(
+      &F, 0, R, 0, 0.0001, 1000, 3, w,
+      &result, &eresult);
+
+    // Clean up
+    gsl_integration_workspace_free(w);
+
+    // Check for errors
+    if (qagstatus != GSL_SUCCESS) {
+      ccl_raise_gsl_warning(qagstatus, "ccl_haloprofile.c: integrate_einasto():");
+      *status = CCL_ERROR_PROFILE_INT;
+      ccl_cosmology_set_status_message(cosmo, "ccl_haloprofile.c: integrate_einasto(): Integration failure\n");
+      return NAN;
+    } else {
+      return result;
+    }
+}
+
+
+//cosmo: ccl cosmology object containing cosmological parameters
+//c: halo concentration, needs to be consistent with halo size definition
+//halomass: halo mass
+//massdef_delta: mass definition, overdensity relative to matter density
+//a: scale factor
+//r: radii at which to calculate output
+//nr: number of radii for calculation
+//rho_r: stores densities at r
+//returns void
+void ccl_halo_profile_einasto(ccl_cosmology *cosmo, double c, double halomass, double massdef_delta_m, double a, double *r, int nr, double *rho_r, int *status){
+
+    //haloradius: halo radius for mass definition
+    //rs: scale radius
+    double haloradius, rs;
+
+    haloradius = r_delta(cosmo, halomass, a, massdef_delta_m, status);
+    rs = haloradius/c;
+
+
+    //nu: peak height, https://arxiv.org/pdf/1401.1216.pdf eqn1
+    //alpha: Einasto parameter, https://arxiv.org/pdf/1401.1216.pdf eqn5
```

Comment:
I think the python-level comment for this function should contain the fact that alpha is set by the mass. This was a choice that you made, rather than letting the user pass in alpha.